### PR TITLE
Added get_path to environment

### PR DESCRIPTION
**Pull request by anki-code**
Hi!

This PR adds `get_path` function to env.

Before:
```python
penv = p"{${...}.get('ENV','/default')}"   # overloaded syntax

if penv.exists():
    ...
```

or
```python
from pathlib import Path                    # overloaded imports
penv = Path(${...}.get('ENV','/default'))   # overloaded calls

if penv.exists():
    ...
```

After:
```python
penv = ${...}.get_path('ENV','/default')   # the clean view

if penv.exists():
    ...
```

### CODE REVIEW

File: xonsh/environ.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 import subprocess
 import platform
 
+from pathlib import Path
```
Comment (scopatz) [inline xonsh/environ.py:17]: This should just be `import pathlib` and grouped with the standard library imports

File: xonsh/environ.py [hunk 2]
```diff
@@ -2002,6 +2003,10 @@ def get(self, key, default=None):
         else:
             return default
 
+    def get_path(self, *args, **kwargs):
```
Comment (scopatz) [inline xonsh/environ.py:2006]: This needs a docstring

**Comment by scopatz** [comment]
Hi @anki-code - thanks for putting this in. I am a little unclear on the semantics here.  How do you know an environment value would be convertable to a path in general? Wouldn't it be better for environment variables to be registered with Path to/from convertors?

**Comment by anki-code** [comment]
> Wouldn't it be better for specific environment variables to be registered with Path to/from convertors?

Don't understand. Give an example please.

File: xonsh/environ.py (hunk 1 above)
Comment (anki-code) [reply xonsh/environ.py]: Fixed

File: xonsh/environ.py (hunk 2 above)
Comment (anki-code) [reply xonsh/environ.py:2006]: Fixed

**Comment by anki-code** [comment]
>  I am a little unclear on the semantics here. How do you know an environment value would be convertable to a path in general?

It's not needed to know. Because the main use case is to get the path from variable and if it's exists (here is the main check) then use it. For example I have docker-compose file that have 3 environment variables that contains paths. The paths in the environment variables is just requirement and if the path exist use it (example in the first message of the thread).

> Wouldn't it be better for specific environment variables to be registered with Path to/from convertors?

Don't understand. Give an example please.

**Comment by scopatz** [comment]
Yeah I guess my point is that in xonsh, environment variables can be of any type (such as dicts) which don't naturally convert straight into Paths. 

> Give an example please.

Like in the default env vars: https://github.com/xonsh/xonsh/blob/master/xonsh/environ.py#L687 or in the down stream projects https://github.com/regro/rever/blob/master/rever/environ.xsh#L268

**Comment by anki-code** [comment]
> Yeah I guess my point is that in xonsh, environment variables can be of any type (such as dicts) which don't naturally convert straight into Paths.

Right. In this case there be an exception and it's absolutely ok (we expect path but get dict/list).

> Like in the default env vars

Thanks! It's interesting! But I see not the same behavior:

```
$ $QWE = '/tmp'
$ ${...}.register(name='QWE', type='path')
$ $QWE
'/tmp'
```
No `path` type here. Why?

I understand that by current logic I should register before use BUT in case with docker I already have environment variables before registration.

**Comment by anki-code** [comment]
Closed this because I should patch and use register.

**Comment by anki-code** [comment]
Closed this because I should patch and use `register` function.

What I should patch:
1. rename `path` type to `env_path`
2. create `path` type as `pathlib.Path`
3. fix getting environment variable when it was set before register

**Pull request closed without merging**